Frameworks/Foundation/NSOperationInternal.h

```diff
@@ -18,4 +18,5 @@
 
 @interface NSOperation ()
 @property (nonatomic, assign, getter=_completionQueue, setter=_setCompletionQueue:) dispatch_queue_t _completionQueue;
+- (BOOL)_acquirePermissionToAddToQueue;
```

Comment:
i like having it as an atomic operation so that it's harder to break contract, i'll try to think of a better name...